src/request.ts

```diff
@@ -938,17 +940,27 @@ class DatastoreRequest {
    *   const apiResponse = data[0];
    * });
    */
-  save(entities, gaxOptions?, callback?) {
+  save(entities: Entities, gaxOptions?: CallOptions): void|Promise<google.datastore.v1.ICommitResponse>;
+  save(entities: Entities, callback: SaveCallback): void;
+  save(entities: Entities, gaxOptionsOrCallback: CallOptions, callback?: SaveCallback): void;
+  save(entities: Entities, gaxOptionsOrCallback?: CallOptions|SaveCallback, cb?: SaveCallback): void|Promise<google.datastore.v1.ICommitResponse> {
     entities = arrify(entities);
 
-    if (is.fn(gaxOptions)) {
-      callback = gaxOptions;
-      gaxOptions = {};
-    }
+    if (typeof gaxOptionsOrCallback === 'function' && typeof gaxOptionsOrCallback !== 'object') {
```

Comment:
same feedback as above.  This logic is a little convoluted, and can be replaced with 2 lines

Reply:
Done!